gossip/src/cluster_info.rs

```diff
@@ -110,6 +110,14 @@ pub const GOSSIP_SLEEP_MILLIS: u64 = 100;
 /// Chosen to be able to handle 1Gbps of pure gossip traffic
 /// 128MB/PACKET_DATA_SIZE
 const MAX_GOSSIP_TRAFFIC: usize = 128_000_000 / PACKET_DATA_SIZE;
+/// Capacity for the [`ClusterInfo::run_socket_consume`] and [`ClusterInfo::run_listen`]
+/// intermediate packet batch buffers.
+///
+/// Uses a heuristic of 28 packets per [`PacketBatch`], which is an observed
+/// average of packets per batch. The buffers are re-used across processing loops,
+/// so any extra capacity that may be reserved due to traffic variations will be preserved,
+/// avoiding excessive resizing and re-allocation.
+const CHANNEL_RECV_BUFFER_INITIAL_CAPACITY: usize = MAX_GOSSIP_TRAFFIC.div_ceil(28);
```

Comment:
Suggest setting this to 32 elements long, and measuring for 24 hours to figure out when does it stop.

Reply:
If this ends up growing to a very huge size, and we keep it resident in memory, we can be wasting gigabytes of RAM doing nothing at all.

Reply:
There are limits on how big this can get:
https://github.com/anza-xyz/agave/blob/2eeb3b5337ff37cba02641aed7ec54f4c09e3adf/gossip/src/cluster_info.rs#L2119